sys/aarch64/context.c

```diff
@@ -59,7 +59,7 @@ int do_setcontext(thread_t *td, ucontext_t *uc) {
   mcontext_t *to = td->td_uctx;
 
   if (uc->uc_flags & _UC_CPU)
-    memcpy(&to->__gregs, &from->__gregs, sizeof(__gregset_t));
+    memcpy(&to->__gregs, &from->__gregs, sizeof(register_t) * (_REG_ELR + 1));
```

Comment:
IIUC you're skipping `SPSR` and `TPIDR`. 

`SPSR` stores both system flags (e.g `DAIF` bits) and user flags (e.g `VCNZ` bits used by comparison instructions). I guess you want to copy whole context but then make sure only user-settable bits can be affected.

Reply:
As you mentioned in private messages [cpu_mcontext_validate](https://grok.dragonflybsd.org/xref/netbsd/sys/arch/aarch64/aarch64/cpu_machdep.c) is what we want.

Reply:
I added validation.

Reply:
Ok.